src/tests/suite/daos_pool.c

```diff
@@ -585,6 +585,371 @@ setup(void **state)
 			  NULL);
 }
 
+/* Private definition for void * typed test_arg_t.pool_lc_args */
+struct test_list_cont {
+	struct test_pool	 tpool;
+	daos_size_t		 nconts;
+	uuid_t			*conts;
+};
+
+static int
+setup_containers(void **state, daos_size_t nconts)
+{
+	test_arg_t		*arg = *state;
+	struct test_list_cont	*lcarg = NULL;
+	int			 i;
+	int			 rc = 0;
+
+	D_ALLOC_PTR(lcarg);
+	if (lcarg == NULL)
+		goto err;
+
+	/***** First, create a pool in which containers will be created *****/
+
+	/* Set some properties in the in/out tpool struct */
+	lcarg->tpool.poh = DAOS_HDL_INVAL;
+	lcarg->tpool.svc.rl_nr = svc_nreplicas;
+	lcarg->tpool.svc.rl_ranks = lcarg->tpool.ranks;
+	lcarg->tpool.pool_size = 1 << 30;	/* 1GB SCM */
+	/* Create the pool */
+	rc = test_setup_pool_create(state, NULL /* ipool */, &lcarg->tpool,
+				    NULL /* prop */);
+	if (rc != 0) {
+		print_message("setup: daos_pool_create failed: %d\n", rc);
+		goto err_free_lcarg;
+	}
+
+	/* TODO: make test_setup_pool_connect() more generic, call here */
+	if (arg->myrank == 0) {
+		rc = daos_pool_connect(lcarg->tpool.pool_uuid, arg->group,
+				       &lcarg->tpool.svc, DAOS_PC_RW,
+				       &lcarg->tpool.poh, NULL /* pool info */,
+				       NULL /* ev */);
+		if (rc != 0)
+			print_message("setup: daos_pool_connect failed: %d\n",
+				      rc);
+	}
+
+	if (arg->multi_rank) {
+		MPI_Bcast(&rc, 1, MPI_INT, 0, MPI_COMM_WORLD);
+		if (rc == 0) {
+			handle_share(&lcarg->tpool.poh, HANDLE_POOL,
+				     arg->myrank, lcarg->tpool.poh, 0);
+		}
+	}
+
+	if (rc != 0)
+		goto err_destroy_pool;
+	print_message("setup: connected to pool: "DF_UUIDF"\n",
+		      DP_UUID(lcarg->tpool.pool_uuid));
+
+	/***** Create many containers in the pool *****/
+
+	if (nconts) {
+		D_ALLOC_ARRAY(lcarg->conts, nconts);
+		if (lcarg->conts == NULL)
+			goto err_disconnect_pool;
```

Comment:
By the way, just FYI, we can just assert that lcarg->conts != NULL and fail the the test otherwise, as the test is not specifically testing D_ALLOC_ARRAY.

Reply:
done